near-sdk-macros/src/core_impl/code_generator/ext.rs

```diff
@@ -0,0 +1,258 @@
+use crate::core_impl::{serializer, AttrSigInfo};
+use proc_macro2::{Ident, TokenStream as TokenStream2};
+use quote::{format_ident, quote, ToTokens};
+use syn::{Generics, Signature};
+
+/// Generates inner ext code for structs and modules. If intended for a struct, generic details
+/// for the struct should be passed in through `generic_details` and the `ext` method will be
+/// added as an impl to the struct ident.
+pub(crate) fn generate_ext_structs(
+    ident: &Ident,
+    generic_details: Option<&Generics>,
+) -> proc_macro2::TokenStream {
+    let name = format_ident!("{}Ext", ident);
+    let mut ext_code = quote! {
+        /// API for calling this contract's functions in a subsequent execution.
+        pub fn ext(account_id: near_sdk::AccountId) -> #name {
+            #name {
+                account_id,
+                amount: 0,
+                static_gas: near_sdk::Gas(0),
+                gas_weight: near_sdk::GasWeight::default(),
+            }
+        }
+    };
+    if let Some(generics) = generic_details {
+        // If ext generation is on struct, make ext function associated with struct not module
+        ext_code = quote! {
+            impl#generics #ident#generics {
+                #ext_code
+            }
+        };
+    }
+
+    quote! {
+      #[must_use]
+      pub struct #name {
+        pub(crate) account_id: near_sdk::AccountId,
+        pub(crate) amount: u128,
+        pub(crate) static_gas: near_sdk::Gas,
+        pub(crate) gas_weight: near_sdk::GasWeight,
+      }
+
+      impl #name {
+        pub fn with_amount(mut self, amount: u128) -> Self {
```

Comment:
Hmm not sure if you forgot to push your changes, but didn't we agree to rename this to `with_deposit_amount`?

Reply:
ah, I was waiting for rebuilding contracts on my laptop and didn't circle back to commit and push, oops. Pushing now